src/Tools/dotnet-monitor/CollectionRules/ActionOptionsDependencyAnalyzer.cs

```diff
@@ -112,18 +115,35 @@ public IList<CollectionRuleActionOptions> GetActionDependencies(int actionIndex)
 
         public object SubstituteOptionValues(IDictionary<string, CollectionRuleActionResult> actionResults, int actionIndex, object settings)
         {
+            //Attempt to substitute context properties.
+            object originalSettings = settings;
+
+            foreach(PropertyInfo propertyInfo in GetPropertiesFromSettings(settings, p => true))
+            {
+                string property = (string)propertyInfo.GetValue(settings);
+                //If we don't have an Endpoint info (such as test scenarios) we cannot perform this substitution.
+                if (string.IsNullOrEmpty(property) || (_ruleContext.EndpointInfo == null))
+                {
+                    continue;
+                }
+                string replacement = property.Replace(RuntimeIdReference, _ruleContext.EndpointInfo.RuntimeInstanceCookie.ToString("D"), StringComparison.Ordinal);
+                if (!ReferenceEquals(replacement, property))
```

Comment:
This is relying on string interning, correct?

Reply:
I am relying on Replace to give me back the original reference if it finds no things to replace. In the case that replacement does happen, the reference would be different than the original regardless of the original.